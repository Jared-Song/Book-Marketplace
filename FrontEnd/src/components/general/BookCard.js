import {
  Card,
  CardActionArea,
  CardMedia,
  Typography,
  makeStyles,
  Grid,
} from "@material-ui/core";
import Rating from "@material-ui/lab/Rating"
import Router from "next/router";
import React from "react";

const useStyles = makeStyles((theme) => ({
  bookImage: {
    width: 160,
    height: 200,
    marginLeft: 20,
  },
  root: {
    width: 200,
    overflow: "hidden",
    marginTop: theme.spacing(1),
    display: "inline-block",
    marginBottom: theme.spacing(1),
  },
  description: {
    fontSize: 12,
    overflow: "hidden",
  },
  contentContainer: {
    height: 80,
    overflow: "hidden",
    padding: 6,
  },
  raitingUserCount: {
    position: "relative",
    bottom: 7,
    left: theme.spacing(1)
  }
}));

export default function BookCard({ book }) {
  const { imageURL, title, price, ratingTotal, ratingNo } = book;
  const classes = useStyles();
  return (
    <Card className={classes.root}>
      <CardActionArea onClick={()=>{
        Router.push("/book/"+book.id);
      }}>
        <CardMedia image={imageURL[0].url} title={title} className={classes.bookImage} />
        <Grid container  className={classes.contentContainer}>
            <Grid item xs={12}>
                <Typography>
                    ${price}    
                </Typography>    
            </Grid>   
            <Grid item xs={12}>
                <Typography noWrap alt={title} className={classes.title}>
                {title}
                </Typography>
            </Grid>
            <Grid item xs={12}>
                <Rating
                  defaultValue={ratingTotal/ratingNo}
                  precision={0.5}
                  readOnly />
                  <Typography variant="caption" className={classes.raitingUserCount}>
                    ({ratingNo})
                  </Typography>
            </Grid>
        </Grid>
      </CardActionArea>
    </Card>
  );
}
