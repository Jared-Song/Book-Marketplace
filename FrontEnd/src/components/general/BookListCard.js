import { Button, Grid, Typography } from "@material-ui/core";
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import BookCard from "./BookCard";
import { lightBlue } from "@material-ui/core/colors";

const useStyles = makeStyles((theme) => ({
  root: {
    // width: "100%",
    backgroundColor: theme.palette.background.paper,
    marginBottom: theme.spacing(2),
  },
  head: {
    backgroundColor: theme.palette.primary.light,
    height: "50px",
    direction: "row",
    justifyContent: "flex-start",
    alignItems: "center",
  },
  title:{
    marginLeft: theme.spacing(1),

    marginRight: theme.spacing(3),
  },
  list: {
    padding: theme.spacing(2),
    direction: "row",
    justifyContent: "space-around",
    alignItems: "left",
  },
}));

export default function BookListCard({ books, title, handleClick }) {
  const classes = useStyles();

  return (
    <Grid container className={classes.root}>
      <Grid item xs={12}>
        <Grid container className={classes.head}>
          <Typography variant="h5" className={classes.title}>{title}</Typography>
          {
            handleClick &&
          <Button onClick={handleClick}>view more {">>"}</Button>

          }
        </Grid>
      </Grid>
      <Grid item xs={12}>
        <Grid container className={classes.list}>
          {books.map((book) => {
            return <BookCard key={book.id} book={book} />;
          })}
        </Grid>
      </Grid>
    </Grid>
  );
}
