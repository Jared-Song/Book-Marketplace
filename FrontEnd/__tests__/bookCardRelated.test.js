import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import BookCard from "../src/components/general/BookCard";
import BookListCard from "../src/components/general/BookListCard";

it("renders bookCard correctly", () => {
  const tree = renderer
    .create(
      <BookCard
        book={{
          id: "1",
          title: "title",
          imageURL: "http://testImage.com",
          price: 1,
          rating: 1,
          ratingNo: 1,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it("renders Book list card correctly", () => {
  const tree = renderer
    .create(
      <BookListCard
        books={[
          {
            id: "1",
            title: "title",
            imageURL: "http://testImage.com",
            price: 1,
            rating: 1,
            ratingNo: 1,
          },
          {
            id: "2",
            title: "2title",
            imageURL: "http://2testImage.com",
            price: 2,
            rating: 2,
            ratingNo: 2,
          },
        ]}
        title="test title"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

