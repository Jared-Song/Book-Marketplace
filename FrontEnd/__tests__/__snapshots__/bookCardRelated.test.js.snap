// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Book list card correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-6 MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root makeStyles-head-7 MuiGrid-container"
    >
      <h5
        className="MuiTypography-root makeStyles-title-8 MuiTypography-h5"
      >
        test title
      </h5>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root makeStyles-list-9 MuiGrid-container"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardMedia-root makeStyles-bookImage-1"
            style={
              Object {
                "backgroundImage": "url(\\"http://testImage.com\\")",
              }
            }
            title="title"
          />
          <div
            className="MuiGrid-root makeStyles-contentContainer-4 MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                $
                1
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                alt="title"
                className="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
              >
                title
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <span
                aria-label="1 Star"
                className="MuiRating-root MuiRating-readOnly"
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
                role="img"
              >
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconFilled"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconFilled"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </span>
              <span
                className="MuiTypography-root makeStyles-raitingUserCount-5 MuiTypography-caption"
              >
                (
                1
                )
              </span>
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="MuiPaper-root MuiCard-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardMedia-root makeStyles-bookImage-1"
            style={
              Object {
                "backgroundImage": "url(\\"http://2testImage.com\\")",
              }
            }
            title="2title"
          />
          <div
            className="MuiGrid-root makeStyles-contentContainer-4 MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                $
                2
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                alt="2title"
                className="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
              >
                2title
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <span
                aria-label="2 Stars"
                className="MuiRating-root MuiRating-readOnly"
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
                role="img"
              >
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconFilled"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconFilled"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconFilled"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconFilled"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  className="MuiRating-decimal"
                >
                  <span
                    style={
                      Object {
                        "overflow": "hidden",
                        "position": "absolute",
                        "width": "0%",
                        "zIndex": 1,
                      }
                    }
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style={Object {}}
                  >
                    <span
                      className="MuiRating-icon MuiRating-iconEmpty"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </span>
              <span
                className="MuiTypography-root makeStyles-raitingUserCount-5 MuiTypography-caption"
              >
                (
                2
                )
              </span>
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders bookCard correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root makeStyles-bookImage-1"
      style={
        Object {
          "backgroundImage": "url(\\"http://testImage.com\\")",
        }
      }
      title="title"
    />
    <div
      className="MuiGrid-root makeStyles-contentContainer-4 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          $
          1
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          alt="title"
          className="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
        >
          title
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <span
          aria-label="1 Star"
          className="MuiRating-root MuiRating-readOnly"
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="img"
        >
          <span
            className="MuiRating-decimal"
          >
            <span
              style={
                Object {
                  "overflow": "hidden",
                  "position": "absolute",
                  "width": "0%",
                  "zIndex": 1,
                }
              }
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span
              style={Object {}}
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            className="MuiRating-decimal"
          >
            <span
              style={
                Object {
                  "overflow": "hidden",
                  "position": "absolute",
                  "width": "0%",
                  "zIndex": 1,
                }
              }
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span
              style={Object {}}
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            className="MuiRating-decimal"
          >
            <span
              style={
                Object {
                  "overflow": "hidden",
                  "position": "absolute",
                  "width": "0%",
                  "zIndex": 1,
                }
              }
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span
              style={Object {}}
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            className="MuiRating-decimal"
          >
            <span
              style={
                Object {
                  "overflow": "hidden",
                  "position": "absolute",
                  "width": "0%",
                  "zIndex": 1,
                }
              }
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span
              style={Object {}}
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            className="MuiRating-decimal"
          >
            <span
              style={
                Object {
                  "overflow": "hidden",
                  "position": "absolute",
                  "width": "0%",
                  "zIndex": 1,
                }
              }
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span
              style={Object {}}
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
        <span
          className="MuiTypography-root makeStyles-raitingUserCount-5 MuiTypography-caption"
        >
          (
          1
          )
        </span>
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
